src/Interpreters/executeQuery.cpp

```diff
@@ -479,12 +219,9 @@ static std::tuple<ASTPtr, BlockIO> executeQueryImpl(
         checkASTSizeLimits(*ast, settings);
 
         /// Put query to process list. But don't put SHOW PROCESSLIST query itself.
-        ProcessList::EntryPtr process_list_entry;
         if (!internal && !ast->as<ASTShowProcesslistQuery>())
         {
-            /// processlist also has query masked now, to avoid secrets leaks though SHOW PROCESSLIST by other users.
-            process_list_entry = context.getProcessList().insert(query_for_logging, ast.get(), context);
-            context.setProcessListElement(&process_list_entry->get());
+            query_process = std::make_shared<QueryProcess>(query_for_logging, ast, internal, context);
```

Comment:
So is `query_process` still only the process list entry, or something more? Because we only have to skip the process list entry for the `SHOW PROCESSLIST` queries, but we still need all other things e.g. the logging context.

Reply:
>So is query_process still only the process list entry, or something more?

QueryProcess encapsulates:
- query logging into log
- query logging into system tables (query_log/query_thread_log)
- query process (entry for system.processes)
- QueryScope (will be configured either from executeQuery() or from the PullingAsyncPipelineExecutor)

> Because we only have to skip the process list entry for the SHOW PROCESSLIST queries, but we still need all other things e.g. the logging context.

AFAICS internal queries logged only to the log, but not to system tables, so they do not require QueryProcess.